extension/module/module.h

```diff
@@ -466,6 +466,32 @@ class Module {
     return set_output("forward", std::move(output_value), output_index);
   }
 
+  /**
+   * EXPERIMENTAL: Updates backend options for a specific method.
+   * Loads the program and method before updating if needed.
+   *
+   * @param[in] method_name The name of the method to update.
+   * @param[in] backend_options The backend options to update the method with.
+   *
+   * @returns An Error to indicate success or failure.
+   */
+  ET_EXPERIMENTAL ET_NODISCARD runtime::Error update(
+      const std::string& method_name,
+      runtime::ArrayRef<runtime::Entry> backend_options);
```

Comment:
I dont think we should expose arrayref options in Module. You should just skip to the syntactic sugar vector/map version you have in the next PR. 

Unrelated to this PR but similar motivations to why I dont want ArrayRef here. I think its probably a mistake we used EValue in module. I wish we had a different type like extension/OwningEValue or something that let us include Dict and List for unflattened IO and let us have stronger ownership semantics on module IO cc @shoumikhin

Reply:
Similarly Im not sure its valuable to have module return Error vs throwing exception